tsl/src/continuous_aggs/create.c

```diff
@@ -722,6 +726,35 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 						 errmsg(
 							 "time bucket function must reference a hypertable dimension column")));
 
+			if (list_length(fe->args) == 3)
+			{
+				/*
+				 * Currently the third argument of the bucketing function can be
+				 * only a timezone. Only immutable expressions can be specified.
+				 */
+				tz_arg = eval_const_expressions(NULL, lthird(fe->args));
+				if (!IsA(tz_arg, Const))
+					ereport(ERROR,
+							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							 errmsg("only immutable expressions allowed in time bucket function"),
+							 errhint("Use an immutable expression as third argument"
+									 " to the time bucket function.")));
+
+				Const *tz = castNode(Const, tz_arg);
+				Assert(tz->consttype == TEXTOID);
+				const char *tz_name = TextDatumGetCString(tz->constvalue);
```

Comment:
What if this is not TEXTOID in a production build? Won't it crash by calling TextDatumGetCString() on a different data type?

Reply:
It shouldn't happen, is_valid_bucketing_function() will take care of it. I will add a corresponding comment.

Reply:
But `ts_is_valid_timezone_name()` is called after `TextDatumGetCString()`.

Reply:
Whoops, ok, wrong function.